stable_baselines/common/vec_env/vec_normalize.py

```diff
@@ -88,48 +89,67 @@ def step_wait(self):
         """
         obs, rews, news, infos = self.venv.step_wait()
-        self.ret = self.ret * self.gamma + rews
         self.old_obs = obs
-        obs = self._normalize_observation(obs)
+        self.old_rews = rews
+
+        if self.norm_obs:
+            if self.training:
+                self.obs_rms.update(obs)
+            obs = self.normalize_obs(obs)
+
         if self.norm_reward:
             if self.training:
-                self.ret_rms.update(self.ret)
-            rews = np.clip(rews / np.sqrt(self.ret_rms.var + self.epsilon), -self.clip_reward, self.clip_reward)
+                self._update_reward(rews)
+            rews = self.normalize_reward(rews)
         self.ret[news] = 0
         return obs, rews, news, infos
 
-    def _normalize_observation(self, obs):
+    def _update_reward(self, rews: np.ndarray) -> None:
+        """Update reward normalizing statistics."""
+        self.ret = self.ret * self.gamma + rews
+        self.ret_rms.update(self.ret)
+
+    def normalize_obs(self, obs: np.ndarray) -> np.ndarray:
         """
-        :param obs: (numpy tensor)
+        Normalize observations using this VecNormalize's observations statistics.
+        Calling this method does not update statistics.
         """
-        if self.norm_obs:
-            if self.training:
-                self.obs_rms.update(obs)
-            obs = np.clip((obs - self.obs_rms.mean) / np.sqrt(self.obs_rms.var + self.epsilon), -self.clip_obs,
-                          self.clip_obs)
-            return obs
-        else:
-            return obs
+        obs = np.clip((obs - self.obs_rms.mean) / np.sqrt(self.obs_rms.var + self.epsilon), -self.clip_obs,
+                      self.clip_obs)
+        return obs
 
-    def get_original_obs(self):
+    def normalize_reward(self, rews: np.ndarray) -> np.ndarray:
         """
-        returns the unnormalized observation
+        Normalize rewards using this VecNormalize's rewards statistics.
+        Calling this method does not update statistics.
+        """
+        rews = np.clip(rews / np.sqrt(self.ret_rms.var + self.epsilon),
+                       -self.clip_reward, self.clip_reward)
+        return rews
 
-        :return: (numpy float)
+    def get_original_obs(self) -> np.ndarray:
+        """
+        Returns an unnormalized version of the observations from the most recent
+        step or reset.
         """
         return self.old_obs
 
+    def get_original_reward(self) -> np.ndarray:
+        """
+        Returns an unnormalized version of the rewards from the most recent step.
+        """
+        return self.old_rews
+
     def reset(self):
         """
         Reset all environments
         """
         obs = self.venv.reset()
-        if len(np.array(obs).shape) == 1:  # for when num_cpu is 1
-            self.old_obs = [obs]
-        else:
-            self.old_obs = obs
+        self.old_obs = obs
```

Comment:
@AdamGleave Shouldn't it be `obs.copy` to avoid modifying `self.old_obs` because here both variables point to the same object?

Reply:
Copying here would be defensive but I don't think is necessary. `self.old_obs` should never get modified?

Reply:
true, if was more afraid of `obs` getting modified outside the wrapper, but I don't remember in which case I had a problem (I think it was with a replay buffer)